lodestar

```diff
@@ -4,4 +4,5 @@
 #
 # ./lodestar.sh beacon --network mainnet
 
+export UV_THREADPOOL_SIZE=$(nproc);
```

Comment:
What about determining and setting this inside js similar to how we determine the worker pool size?

It could be set here maybe, or some other preload script in the CLI package
https://github.com/ChainSafe/lodestar/blob/b81230acea876feb60b2bc5f530e1f486ce332e5/packages/cli/src/applyPreset.ts#L9

We only need to set UV_THREADPOOL_SIZE env before any task involving the UV threadpool is executed, ie. reading a file via async api

Reply:
Thats a good idea.  It would literally need to be the first line of the handler.... `beaconHandlerInit` does `fs` to setup the env.  

<img width="871" alt="Screenshot 2024-04-03 at 11 11 48 AM" src="https://github.com/ChainSafe/lodestar/assets/18608739/a1f3521a-bbf7-40cb-b9ae-d5b1fc88ae53">


I also kind of like the idea of giving such a critical setting more visibility like is the case with `--max-old-space-size=8192`.   I kinda like it out in the startup file but definitely see the allure of baking it in.  Had to redeploy a feat group because i added it to the script but we dont use the script so it got left out.  Guessing that will be the same for other operators as well.

Curious what does everyone else think??

Reply:
> we dont use the script so it got left out. Guessing that will be the same for other operators as well.

Yes, you can't assume they use the helper script or even our docker image in some cases. The good thing with max old space is that it hard fails if not set, while not setting this might still work but performance is degraded.

Reply:
Updated @nflaig.  Great idea btw!!!  🥷